tests/periph_spi/main.c

```diff
@@ -242,163 +267,236 @@ int cmd_bench(int argc, char **argv)
+                                                        sched_start));
 
     /* 9 - read 1000 times 2 byte from register */
+    sched_start = _sched_ticks();
     start = xtimer_now_usec();
     for (int i = 0; i < BENCH_REDOS; i++) {
         spi_transfer_regs(spiconf.dev, spiconf.cs, BENCH_REGADDR,
                           NULL, bench_rbuf, BENCH_SMALL);
     }
     stop = xtimer_now_usec();
+    sched_stop = _sched_ticks();
     printf(" 9 - read %i times %i byte from register:", BENCH_REDOS, BENCH_SMALL);
-    printf("\t%i\n", (int)(stop - start));
+    printf("\t%i\t%i\n", (int)(stop - start),
+           (int)xtimer_diff(sched_stop, sched_start).ticks32);
     sum += (stop - start);
+    sched_sum += xtimer_usec_from_ticks(xtimer_diff(sched_stop,
+                                                        sched_start));
 
     /* 10 - read 1000 times 100 byte from register */
+    sched_start = _sched_ticks();
     start = xtimer_now_usec();
     for (int i = 0; i < BENCH_REDOS; i++) {
         spi_transfer_regs(spiconf.dev, spiconf.cs, BENCH_REGADDR,
                           NULL, bench_rbuf, BENCH_LARGE);
     }
     stop = xtimer_now_usec();
+    sched_stop = _sched_ticks();
     printf("10 - read %i times %i byte from register:", BENCH_REDOS, BENCH_LARGE);
-    printf("\t%i\n", (int)(stop - start));
+    printf("\t%i\t%i\n", (int)(stop - start),
+           (int)xtimer_diff(sched_stop, sched_start).ticks32);
     sum += (stop - start);
+    sched_sum += xtimer_usec_from_ticks(xtimer_diff(sched_stop,
+                                                        sched_start));
 
     /* 11 - transfer 1000 times 2 byte */
+    sched_start = _sched_ticks();
     start = xtimer_now_usec();
     for (int i = 0; i < BENCH_REDOS; i++) {
         spi_transfer_bytes(spiconf.dev, spiconf.cs, false,
                            bench_wbuf, bench_rbuf, BENCH_SMALL);
     }
     stop = xtimer_now_usec();
+    sched_stop = _sched_ticks();
     printf("11 - transfer %i times %i byte:", BENCH_REDOS, BENCH_SMALL);
-    printf("\t\t%i\n", (int)(stop - start));
+    printf("\t\t%i\t%i\n", (int)(stop - start),
+           (int)xtimer_diff(sched_stop, sched_start).ticks32);
     sum += (stop - start);
+    sched_sum += xtimer_usec_from_ticks(xtimer_diff(sched_stop,
+                                                        sched_start));
 
     /* 12 - transfer 1000 times 100 byte */
+    sched_start = _sched_ticks();
     start = xtimer_now_usec();
     for (int i = 0; i < BENCH_REDOS; i++) {
         spi_transfer_bytes(spiconf.dev, spiconf.cs, false,
                            bench_wbuf, bench_rbuf, BENCH_LARGE);
     }
     stop = xtimer_now_usec();
+    sched_stop = _sched_ticks();
     printf("12 - transfer %i times %i byte:", BENCH_REDOS, BENCH_LARGE);
-    printf("\t\t%i\n", (int)(stop - start));
+    printf("\t\t%i\t%i\n", (int)(stop - start),
+           (int)xtimer_diff(sched_stop, sched_start).ticks32);
```

Comment:
I know `%"PRIu32"` is ugly and just casting to `int` is much easier, but on ATmega this will print

```
2020-05-18 22:52:35,723 # ### Running some benchmarks, all values in [us] ###
2020-05-18 22:52:35,757 # ### Test				Transfer time	user time
2020-05-18 22:52:35,757 # 
2020-05-18 22:52:35,825 #  1 - write 1000 times 1 byte:			24776	6206
2020-05-18 22:52:35,900 #  2 - write 1000 times 2 byte:			31272	7830
2020-05-18 22:52:36,939 #  3 - write 1000 times 100 byte:		10272	-13803
2020-05-18 22:52:37,041 #  4 - write 1000 times 1 byte to register:	-19136	11613
2020-05-18 22:52:37,150 #  5 - write 1000 times 2 byte to register:	-11080	13627
2020-05-18 22:52:38,224 #  6 - write 1000 times 100 byte to register:	-32052	-8000
2020-05-18 22:52:38,297 #  7 - read 1000 times 2 byte:			31396	7862
2020-05-18 22:52:39,329 #  8 - read 1000 times 100 byte:			4200	-15322
2020-05-18 22:52:39,441 #  9 - read 1000 times 2 byte from register:	-11076	13628
2020-05-18 22:52:40,508 # 10 - read 1000 times 100 byte from register:	27344	-9535
2020-05-18 22:52:40,584 # 11 - transfer 1000 times 2 byte:		31400	7862
2020-05-18 22:52:41,633 # 12 - transfer 1000 times 100 byte:		16612	-12219
2020-05-18 22:52:41,746 # 13 - transfer 1000 times 2 byte to register:	-10948	13659
2020-05-18 22:52:42,829 # 14 - transfer 1000 times 100 byte to register:	-25816	-6441
2020-05-18 22:52:43,859 # -- - SUM:					1628	6359324
2020-05-18 22:52:43,859 # 
2020-05-18 22:52:43,886 # ### All runs complete ###
```

Reply:
Good point! I've fixed this for the second column. I suggest we fix the first column in a follow up.

Reply:
Why not also fix the first column while you're at it?

Reply:
I can change it here in a separate commit, but I don't want to mix things here too much

Reply:
Yea it's good to have consistency (and proper output)

Reply:
Added a commit

Reply:
Looks good now

```
2020-05-19 11:09:32,410 # ### Running some benchmarks, all values in [us] ###
2020-05-19 11:09:32,444 # ### Test				Transfer time	user time
2020-05-19 11:09:32,444 # 
2020-05-19 11:09:32,521 #  1 - write 1000 times 1 byte:			32768	8205
2020-05-19 11:09:32,612 #  2 - write 1000 times 2 byte:			47268	11829
2020-05-19 11:09:34,453 #  3 - write 1000 times 100 byte:		1793584	448408
2020-05-19 11:09:34,570 #  4 - write 1000 times 1 byte to register:	62332	15595
2020-05-19 11:09:34,703 #  5 - write 1000 times 2 byte to register:	78400	19612
2020-05-19 11:09:36,587 #  6 - write 1000 times 100 byte to register:	1824772	456205
2020-05-19 11:09:36,677 #  7 - read 1000 times 2 byte:			47340	11847
2020-05-19 11:09:38,513 #  8 - read 1000 times 100 byte:			1787504	446888
2020-05-19 11:09:38,646 #  9 - read 1000 times 2 byte from register:	78460	19628
2020-05-19 11:09:40,526 # 10 - read 1000 times 100 byte from register:	1818688	454685
2020-05-19 11:09:40,619 # 11 - transfer 1000 times 2 byte:		47400	11862
2020-05-19 11:09:42,470 # 12 - transfer 1000 times 100 byte:		1799880	449982
2020-05-19 11:09:42,606 # 13 - transfer 1000 times 2 byte to register:	78520	19643
2020-05-19 11:09:44,499 # 14 - transfer 1000 times 100 byte to register:	1831084	457783
2020-05-19 11:09:45,529 # -- - SUM:					11328000	11328688
2020-05-19 11:09:45,561 # 
2020-05-19 11:09:45,561 # ### All runs complete ###
```

Reply:
@benpicco Something is wrong with that second column though, those numbers don't add up to the resulting sum…

Reply:
I wonder if that's an atmega specific, maybe in the xtimer_diff code.